### PR TITLE
ENH dark mode

### PR DESCRIPTION
**Pull request by andrewandante**
image

Should probably save this for Hacktober but YOLO

## Issues
- https://github.com/silverstripe/github-issue-search-client/issues/114

### CODE REVIEW

**Comment by GuySartorelli** [comment]
Sorry, this PR is gonna need to be on hold for a hot minute - Vercel is upset that we're using an ancient version of node so I'll need to resolve that before I can properly review this PR.

**Comment by GuySartorelli** [comment]
Vercel is happy again. Can you please rebase the PR?

**Comment by GuySartorelli** [comment]
Vercel is unhappy with the dependencies for some reason - I've blown away the lock file and re-run `npm install` and hoping that satisfies it.

File: src/App.vue [hunk 1]
```diff
@@ -23,19 +24,29 @@
...
 });
 </script>
 
 <style>
+  html.dark {
+    color-scheme: dark;
+  }
+
+  .dark {
+    color: #31a8ff;
```
Comment (GuySartorelli) [inline src/App.vue:47]: I assume this is where the text is getting its blue from. Can we please swap this out for a nice light grey? My brain is wired to think "blue means link" and now suddenly everything on the page looks like a link to me.

File: src/components/DarkModeToggle.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+<template>
+<vue-toggle
+    :title="isDark ? 'Dark Mode' : 'Light Mode'"
+    name="Toggle Dark"
+    @click="toggleDark()"
+    class="dark-mode-toggle"
+    :class="{ dark: isDark }"
+    :toggled="isDark"
+    activeColor="#31a8ff"
```
Comment (GuySartorelli) [inline src/components/DarkModeToggle.vue:17]: Is this blue only used for the toggle button itself? Or is it somehow connected to the blue in the `<style>` element in App.vue?

File: src/App.vue (hunk 1 above)
Comment (andrewandante) [reply src/App.vue:47]: Yep, that's the main "text" colour - will opt for something light gray and with high-enough contrast

File: src/components/DarkModeToggle.vue (hunk 2 above)
Comment (andrewandante) [reply src/components/DarkModeToggle.vue:17]: This one is purely the toggle, yeah. Defaults to green otherwise 🤢

File: src/App.vue (hunk 1 above)
Comment (andrewandante) [reply src/App.vue]: Updated screenshot and PR 👍

**Review by GuySartorelli — APPROVED** [review]
Looks nice, thanks!

**Comment by n8-dev** [comment]
Bahaha, I literally thought of this for hacktober @andrewandante 👏 👏 👏 👏 👏

**Pull request merged**